src/clojure/nrepl/transport.clj

```diff
@@ -79,7 +80,7 @@
      (catch EOFException e#
        (throw (SocketException. "The transport's socket appears to have lost its connection to the nREPL server")))
      (catch Throwable e#
-       (if (and ~s (not (.isConnected ~s)))
+       (if (and ~s (not (.isClosed ~s)))
```

Comment:
I don't understand the need for this change. What's the motivation for it?

Reply:
At `nrepl.core-test`, the tests `clients-fail-on-disconnects` and `transports-fail-on-disconnects` failswhen using `.isConnected` gives me an error, both using the transit transport with messagepack.

Reply:
Sorry, after changing the implementation, I saw that it was not affecting the tests, so it will be the same as before